File: VB/CustomHandlers/SchedulerMoveEventHandler.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports DevExpress.XtraScheduler
Imports DevExpress.XtraScheduler.Drawing
Imports System.Drawing
Imports System.Windows.Forms
Imports DevExpress.Xpf.Scheduler
Imports DevExpress.XtraScheduler.Native
Imports DevExpress.Xpf.Scheduler.Drawing
Imports System.Windows
Imports System.Windows.Media

Public MustInherit Class SchedulerMoveEventHandler
	Protected control As SchedulerControl

	Protected ReadOnly Property MousePosition() As System.Windows.Point
		Get
			Dim currentPosition As System.Drawing.Point = Form.MousePosition
			Return control.PointFromScreen(New System.Windows.Point(currentPosition.X, currentPosition.Y))
		End Get
	End Property
	Protected ReadOnly Property HitTimeCellInfo() As HitTimeCellInfo
		Get
			Return HitTimeCellInfo.CreateHitTimeCellInfo(control, MousePosition)
		End Get
	End Property

	Protected ReadOnly Property AppointmentViewInfo() As IAppointmentView
		Get
			Return GetAppointmentViewInfo()
		End Get
	End Property

	Protected ReadOnly Property SelectableIntervalViewInfo() As ISelectableIntervalViewInfo
		Get
			Return GetSelectableIntervalViewInfo()
		End Get
	End Property

	Public Overridable Sub AttachToControl()
	End Sub
	Public Overridable Sub DetachFromControl()
	End Sub

	Protected Sub New(ByVal control As SchedulerControl)
		Me.control = control
	End Sub

	Protected Function GetAppointmentViewInfo() As IAppointmentView
		Dim hitInfo As SchedulerControlHitInfo = control.ActiveView.CalcHitInfo(MousePosition)
		Return If((hitInfo.HitTest = SchedulerHitTest.AppointmentContent), CType(hitInfo.ViewInfo, IAppointmentView), Nothing)
	End Function

	Protected Function GetSelectableIntervalViewInfo() As ISelectableIntervalViewInfo
		Dim hitInfo As SchedulerControlHitInfo = control.ActiveView.CalcHitInfo(MousePosition)
		Return If((hitInfo.HitTest = SchedulerHitTest.Cell), CType(hitInfo.ViewInfo, ISelectableIntervalViewInfo), Nothing)
	End Function
End Class

File: CS/CustomHandlers/SchedulerMoveEventHandler.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using DevExpress.XtraScheduler;
using DevExpress.XtraScheduler.Drawing;
using System.Drawing;
using System.Windows.Forms;
using DevExpress.Xpf.Scheduler;
using DevExpress.XtraScheduler.Native;
using DevExpress.Xpf.Scheduler.Drawing;
using System.Windows;
using System.Windows.Media;

public abstract class SchedulerMoveEventHandler {
    protected SchedulerControl control;

    protected System.Windows.Point MousePosition {
        get { 
            System.Drawing.Point currentPosition = Form.MousePosition;
            return control.PointFromScreen(new System.Windows.Point(currentPosition.X, currentPosition.Y)); 
        }
    }
    protected HitTimeCellInfo HitTimeCellInfo {
        get { return HitTimeCellInfo.CreateHitTimeCellInfo(control, MousePosition); }
    }

    protected IAppointmentView  AppointmentViewInfo {
        get { return GetAppointmentViewInfo(); }
    }

    protected ISelectableIntervalViewInfo SelectableIntervalViewInfo {
        get { return GetSelectableIntervalViewInfo(); }
    }

    public virtual void AttachToControl() { }
    public virtual void DetachFromControl() { }

    protected SchedulerMoveEventHandler(SchedulerControl control) {
        this.control = control;
    }
    
    protected IAppointmentView  GetAppointmentViewInfo() {
        SchedulerControlHitInfo hitInfo = control.ActiveView.CalcHitInfo(MousePosition);
        return (hitInfo.HitTest == SchedulerHitTest.AppointmentContent) ? (IAppointmentView )hitInfo.ViewInfo : null;
    }

    protected ISelectableIntervalViewInfo GetSelectableIntervalViewInfo() {
        SchedulerControlHitInfo hitInfo = control.ActiveView.CalcHitInfo(MousePosition);
        return (hitInfo.HitTest == SchedulerHitTest.Cell) ? (ISelectableIntervalViewInfo)hitInfo.ViewInfo : null;
    }
}